.github/workflows/portal-tests.yml

```diff
@@ -0,0 +1,38 @@
+name: Portal Unit Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Change directory to portal
+        run: cd portal
+
+      - uses: pnpm/action-setup@v4
+        name: Install pnpm
+        with:
+          version: 8
+          run_install: false
+
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Install dependencies
+        working-directory: portal
```

Comment:
Question: why `working-directory: portal` here, while elsewhere there is `cd portal`?

Reply:
Ah that was a remnant while I was testing the difference between the two. 
`working-directory` should be used because it has no side effects on next steps compared to `cd`.

Reply:
Ok, up to you if this needs further investigation, or if the result works.